addons/binding/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/handler/VelbusThingHandler.java

```diff
@@ -0,0 +1,174 @@
+            } else {
+                subAddresses[i] = (byte) 0xFF;
+            }
+        }
+
+        return new VelbusModuleAddress(address, numberOfSubAddresses);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing velbus handler.");
+        Bridge bridge = getBridge();
+        initializeThing(bridge == null ? null : bridge.getStatus());
+        initializeChannelNames();
+        initializeChannelStates();
+    }
+
+    protected VelbusModuleAddress getModuleAddress() {
+        return velbusModuleAddress;
+    }
+
+    @SuppressWarnings("null")
+    protected void updateChannelLabel(ChannelUID channelUID, String channelName, String acceptedItemType) {
+        if (channelUID != null && channelName != null) {
+            Channel existingChannel = thing.getChannel(channelUID.getId());
+            if (existingChannel != null) {
+                String acceptedItem = existingChannel.getAcceptedItemType();
+                Configuration configuration = existingChannel.getConfiguration();
+                Set<String> defaultTags = existingChannel.getDefaultTags();
+                String description = existingChannel.getDescription();
+                ChannelKind kind = existingChannel.getKind();
+                Map<String, String> properties = existingChannel.getProperties();
+                ChannelTypeUID type = existingChannel.getChannelTypeUID();
+
+                ThingBuilder thingBuilder = editThing();
+                Channel channel = ChannelBuilder.create(channelUID, acceptedItem).withConfiguration(configuration)
+                        .withDefaultTags(defaultTags).withDescription(description != null ? description : "")
+                        .withKind(kind).withLabel(channelName).withProperties(properties).withType(type).build();
+                thingBuilder.withoutChannel(channelUID).withChannel(channel);
+                updateThing(thingBuilder.build());
+            }
+        }
+    }
+
+    private void initializeThing(ThingStatus bridgeStatus) {
+        logger.debug("initializeThing thing {} with address {} bridge status {}", getThing().getUID(),
+                velbusModuleAddress.getAddress(), bridgeStatus);
+
+        // note: this call implicitly registers our handler as a listener on
+        // the bridge
+        if (getVelbusBridgeHandler() != null) {
+            if (bridgeStatus == ThingStatus.ONLINE) {
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE);
+        }
+    }
+
+    private void initializeChannelNames() {
+        for (int i = 1; i <= numberOfChannels; i++) {
+            String propertyKey = CHANNEL + i;
+            if (getThing().getProperties().containsKey(propertyKey)) {
+                String channelName = getThing().getProperties().get(propertyKey);
```

Comment:
The properties weren't required when the things were specified via text files, because a default channel name "CHx" was used. However, I modified it to allowed specification of the channel name via text file as well.